### PR TITLE
Vendoring directed tests

### PR DESCRIPTION
**Pull request by Saad525**
Using the directed test UVM flow, vendoring the directed tests and adding testlists for riscv-tests, riscv-arch-tests and ePMP tests.

Vendored repositories:
- riscv-tests
- riscv-test-env
- riscv-arch-tests
- riscv-isa-sim (for ePMP tests)

Modifications to vendored repositories:
- start label address, 0x80000080 (patch)
- test ending routine
- ePMP tests specific (forked repo)
  - removing redundant prints
  - pmp permissions for signature address

Script:
- gen_testlist.py to populate directed_testlist.yaml
  - adds custom directed tests
  - adds directed tests from vendored repositories (depending on supported extensions)

### CODE REVIEW

File: dv/uvm/core_ibex/directed_tests/READ.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Directed Tests
```
Comment (GregAC) [inline dv/uvm/core_ibex/directed_tests/READ.md]: Why did you remove the README in the next commit? If you don't want to include it as part of the PR please just drop the commits entirely.

Comment (GregAC) [reply dv/uvm/core_ibex/directed_tests/READ.md]: Oh and please rename to README.md if you do keep it.

File: vendor/riscv_isa_sim.lock.hjson [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+// This file is generated by the util/vendor script. Please do not modify it
+// manually.
+
+{
+  upstream:
+  {
+    url: https://github.com/joxie/riscv-isa-sim
+    rev: c2186bf1731b2a123ccc785ce9585861d370886f
```
Comment (GregAC) [inline vendor/riscv_isa_sim.lock.hjson]: As we only want the tests out of this and nothing else let's just vendor in those and cut out the rest. This is what we do with the 'lowrisc_ip' directory which is actually vendoring in part of OpenTitan see https://github.com/lowRISC/ibex/blob/master/vendor/lowrisc_ip.vendor.hjson to see how it works.

**Review by GregAC — CHANGES_REQUESTED** [review]
Looking good but a few things to address around how we work with vendoring before we can merge this.

For the commit removing a bunch of files from riscv-arch-test note the vendoring tool can handle this: https://docs.opentitan.org/doc/ug/vendor_hw/#how-to-exclude-some-files-from-the-upstream-repository so please use that method.

File: vendor/riscv-test-env/v/riscv_test.h [hunk 3]
```diff
@@ -24,6 +24,16 @@
 extra_boot:                                                             \
         EXTRA_INIT                                                      \
         ret;                                                            \
+.global trap_filter;                                                    \
```
[the remark this answers is not in the document]
Comment (Saad525) [reply vendor/riscv-test-env/v/riscv_test.h:27]: These changes were actually from the latest upstream repo, so I updated the commit hash too when made this change.

File: vendor/riscv-test-env/v/vm.c [hunk 4]
```diff
@@ -136,8 +136,14 @@ static void evict(unsigned long addr)
   }
 }
 
+extern int pf_filter(uintptr_t addr, uintptr_t *pte, int *copy);
```
[the remark this answers is not in the document]
Comment (Saad525) [reply vendor/riscv-test-env/v/vm.c:139]: Same as that of `vendor/riscv-test-env/v/riscv_test.h`

File: vendor/riscv_isa_sim.lock.hjson (hunk 2 above)
Comment (Saad525) [reply vendor/riscv_isa_sim.lock.hjson]: Thanks for pointing this. I have updated this to only include the needed files/directories.

```
tests/mseccfg
arch_test_target
LICENSE
```

File: dv/uvm/core_ibex/directed_tests/READ.md (hunk 1 above)
Comment (Saad525) [reply dv/uvm/core_ibex/directed_tests/READ.md]: There was a documentation check failing so removed it temporarily. Now added, probably it was failing due to the naming typo.

File: dv/uvm/core_ibex/directed_tests/directed_testlist.yaml [hunk 5]
```diff
@@ -25,16 +26,6440 @@
...
 
 - test: access_pmp_overlap
   desc: >
     PMP access basic test
   iterations: 1
   test_srcs: access_pmp_overlap/access_pmp_overlap.S
-  config: base
+  config: riscv-tests
+
+# Test-suite: riscv-tests
```
[the remark this answers is not in the document]
Comment (Saad525) [reply dv/uvm/core_ibex/directed_tests/directed_testlist.yaml:70]: Yes, that would be useful. Subset based on configs maybe? as it is a sort of current distribution in testlist.

**Review by GregAC — APPROVED** [review]
